src/core/output/outputSplitter.ts

```diff
@@ -0,0 +1,53 @@
+import type { ProcessedFile } from '../file/fileTypes.js';
+import { TokenCounter } from '../tokenCount/tokenCount.js';
+
+export interface OutputSplit {
+  partNumber: number;
+  tokenCount: number;
+  includedFiles: ProcessedFile[]; // Add includedFiles property
+}
+
+export const splitOutput = (
+  processedFiles: ProcessedFile[],
+  maxTokensPerFile: number,
+): OutputSplit[] => {
+  const tokenCounter = new TokenCounter();
+  const outputSplits: OutputSplit[] = [];
+  let currentTokenCount = 0;
+  let currentOutput = '';
+  let currentIncludedFiles: ProcessedFile[] = []; // Initialize currentIncludedFiles
+
+  for (const file of processedFiles) {
+    const fileTokenCount = tokenCounter.countTokens(file.content, file.path);
```

Comment:
If there is a large number of files, the heavy processing of token count may be affected. Currently, the parts where token counts are being performed are slightly mitigated by adding sleep.

However, I thought it would be better to leave it as it is for now and fix it when the problem actually occurs.